src/main/java/seedu/address/model/student/Student.java

```diff
@@ -27,23 +31,68 @@
     private final Year year;
     private final Admin admin;
     private final List<Question> questions = new ArrayList<>();
-    private final List<Exam> exams = new ArrayList<>();
     private final Academic academic;
 
     /**
-     *  name, phone, school, year, must be present and not null.
-     *  exams is empty when a student is first initialised.
+     *  alternate constructor where admin and academic details need not be changed
      */
     public Student(Name name, Phone phone, School school, Year year,
-                   Admin admin, List<Question> questions, List<Exam> exams, Academic academic) {
-        requireAllNonNull(name, phone, school, year, admin, academic);
+                   Admin admin, List<Question> questions, Academic academic) {
+        requireAllNonNull(name, phone, school, year, admin, questions, academic);
         this.name = name;
         this.phone = phone;
         this.school = school;
         this.year = year;
         this.admin = admin;
         this.questions.addAll(questions);
-        this.exams.addAll(exams);
+        this.academic = academic;
+    }
+
+    /**
+     * default constructor for when all fields are added
+     */
+    public Student(Name name, Phone phone, School school, Year year,
+                   ClassVenue venue, ClassTime time, Fee fee, PaymentDate date, List<Detail> details,
+                   List<Question> questions, List<Exam> exams, List<Attendance> attendances) {
+        requireAllNonNull(name, phone, school, year, venue, time, fee, date, details, questions, exams, attendances);
+        this.name = name;
+        this.phone = phone;
+        this.school = school;
+        this.year = year;
+        this.admin = new Admin(venue, time, fee, date, details);
+        this.questions.addAll(questions);
+        this.academic = new Academic(attendances, exams);
+    }
+
+    /**
+     * alternate constructor for when admin need not be changed
+     */
+    public Student(Name name, Phone phone, School school, Year year,
+                   Admin admin, List<Question> questions,
+                   List<Exam> exams, List<Attendance> attendances) {
+        requireAllNonNull(name, phone, school, year, admin, questions, exams, attendances);
+        this.name = name;
+        this.phone = phone;
+        this.school = school;
+        this.year = year;
+        this.admin = admin;
+        this.questions.addAll(questions);
+        this.academic = new Academic(attendances, exams);
+    }
+
+    /**
+     * alternate constructor for when academic need not be changed
+     */
+    public Student(Name name, Phone phone, School school, Year year,
+                   ClassVenue venue, ClassTime time, Fee fee, PaymentDate date, List<Detail> details,
+                   List<Question> questions, Academic academic) {
```

Comment:
Should we add questions into academics as well? Because the GUI currently displays questions as an academic type.

Reply:
I'll handle it in another PR